backend/core/llm_chain.py

```
@@ -3,490 +3,317 @@
-                    del _response_cache[cache_key]
-            else:
-                # Remove expired entry
-                del _response_cache[cache_key]
-                logger.debug("Removed expired cache entry")
-
-    except Exception as e:
-        logger.error(f"Error retrieving cached response: {e}")
-        # Remove potentially corrupted cache entry
-        if cache_key in _response_cache:
+    """Retrieves a final response from the cache if available and not expired."""
+    if not cache_key or not ENABLE_CACHING: return None
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)